src/dials/algorithms/correlation/analysis.py

```diff
@@ -216,7 +230,7 @@ def calculate_matrices(self):
             dims_to_test = self.params.dimensionality_assessment.maximum_dimensions
 
         if self.params.dimensions is Auto:
-            self.cosym_analysis._determine_dimensions(
+            self._dim, self._func = self.cosym_analysis._determine_dimensions(
```

Comment:
If dimensions is not Auto, then we have a failure in L476 when calling the `plot_dims` function as `self._dim` and `self._func` are not defined.